File: app/src/main/java/io/agora/chat/uikit/simple/SignInActivity.java
package io.agora.chat.uikit.simple;

import android.os.Bundle;
import android.view.View;
import android.widget.Toast;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;

import io.agora.CallBack;
import io.agora.chat.ChatClient;
import io.agora.chat.uikit.simple.databinding.ActivitySignInBinding;

public class SignInActivity extends AppCompatActivity {
    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        ActivitySignInBinding binding = ActivitySignInBinding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());

        if(ChatClient.getInstance().getOptions().getAutoLogin() && ChatClient.getInstance().isLoggedInBefore()) {
            MainActivity.actionStart(SignInActivity.this, "");
            finish();
        }else {
            binding.btnLogin.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View v) {
                    String username = binding.etUsername.getText().toString().trim();
                    String trim = binding.etPassword.getText().toString().trim();
                    String toChatName = binding.etToChatUsername.getText().toString().trim();
                    ChatClient.getInstance().login(username, trim, new CallBack() {
                        @Override
                        public void onSuccess() {
                            MainActivity.actionStart(SignInActivity.this, toChatName);
                            finish();
                        }

                        @Override
                        public void onError(int i, String s) {
                            runOnUiThread(()->Toast.makeText(SignInActivity.this, "error: "+i+" error message: "+s, Toast.LENGTH_SHORT).show());
                        }

                        @Override
                        public void onProgress(int i, String s) {

                        }
                    });
                }
            });
        }

    }
}

File: app/src/main/java/io/agora/chat/uikit/simple/SignInActivity.kt
package io.agora.chat.uikit.simple

import android.content.Intent
import android.os.Bundle
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import io.agora.chat.uikit.simple.databinding.ActivitySignInBinding
import io.agora.uikit.EaseIM
import io.agora.uikit.common.ChatClient
import io.agora.uikit.common.extensions.showToast

class SignInActivity : AppCompatActivity() {

    private val binding: ActivitySignInBinding = ActivitySignInBinding.inflate(layoutInflater)
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(binding.root)

        if(ChatClient.getInstance().options.autoLogin && ChatClient.getInstance().isLoggedInBefore) {
            startActivity(Intent(this,SignInActivity::class.java))
            finish()
        }else{

            binding.btnLogin.setOnClickListener{
                val username = binding.etUsername.text.toString().trim()
                val trim = binding.etPassword.text.toString().trim()
                val toChatName = binding.etToChatUsername.text.toString().trim()
                EaseIM.login(username,trim,
                    onSuccess = {
                        MainActivity.actionStart(this,toChatName)
                    },
                    onError = { code, error ->
                        runOnUiThread{
                            showToast("code: $code  error: $error")
                        }
                    })
            }

        }
    }

}